Assets/Scripts/Ball.cs

```diff
@@ -4,15 +4,45 @@
 
 public class Ball : MonoBehaviour
 {
-    // Start is called before the first frame update
+    public float speed = 10;
+
+    private Rigidbody rb;
+    private Vector3 lastFrameVelocity;
+
     void Start()
     {
-        
+        rb = GetComponent<Rigidbody>();
```

Comment:
Getcomponents are good on awake if the component is on itself that way you dont need to rewrite it every time on Enable, ok here for velocity.